CMakeLists.txt

```diff
@@ -12,6 +12,30 @@
 #=========================================================================================
 
 cmake_minimum_required(VERSION 3.12)
+
+# Load machine specific defaults such as architecture and mpi launch command.
+# Command line argument takes precedence over environment variable.
+# Loading this before project definition to allow setting the compiler.
+if (MACHINE_CFG)
+  if(EXISTS "${MACHINE_CFG}")
+    include(${MACHINE_CFG})
+  else()
+    message(FATAL_ERROR "Given machine configuration at "
+      "${MACHINE_CFG} not found.")
```

Comment:
I like that it dies if you provide a missing file. I think this will be useful for debugging.